### PR TITLE
Add static-website templates

### PR DESCRIPTION
**Pull request by cnunciato**
This is a first pass at creating a set of static-website templates for AWS, Azure, and Google Cloud in TypeScript, Python, Go, C#, and YAML. Includes some canned web content, buckets configured for web access, and basic CDN configuration.

These can all be tested now by running:

```bash
pulumi new https://github.com/pulumi/templates/tree/cnunciato/static-website/{template-folder-name}
``` 

Fixes https://github.com/pulumi/devrel-team/issues/128.

### CODE REVIEW

File: static-website-aws-csharp/Program.cs
Comment (joeduffy) [inline static-website-aws-csharp/Program.cs:9]: Should these be using the `aws:region` config?

File: static-website-aws-go/main.go
Comment (joeduffy) [inline static-website-aws-go/main.go:49]: Might be slightly nicer using the `pulumi.Sprintf` function?

File: static-website-aws-csharp/Program.cs
Comment (cnunciato) [reply static-website-aws-csharp/Program.cs:9]: Indeed — sure what happened there, maybe something amiss with `pulumi convert`. Will take a look!

File: static-website-aws-go/main.go
Comment (cnunciato) [reply static-website-aws-go/main.go]: Absolutely. These are all still the generated code from `pulumi convert’, but I plan to add some scripting to tidy things up.

**Comment by aaronkao** [comment]
A piece of feedback while talking to Christian earlier

A distinction should be made between project config options, infrastructure config options, and the app level config options (e.g., index.html as default index document or the path)in the pulumi new project wizard in the CLI

File: static-website-aws-csharp/Program.cs
Comment (cnunciato) [reply static-website-aws-csharp/Program.cs]: Fixed!

**Comment by cnunciato** [comment]
All templates are tested and usable at this point, but the code for each one isn't quite where I'd like to be yet. (Still all generated from `pulumi convert` with a handful of fixes.) Will take care of that next.

**Comment by cnunciato** [comment]
I believe the test failures here are unrelated to the content of the templates. @t0yv0 @stack72 is this something the platform team is still looking investigating?

File: static-website-aws-csharp/Program.cs
Comment (cnunciato) [inline static-website-aws-csharp/Program.cs:17]: I'll run through and add some enums as well for the languages that have them for these.

Comment (cnunciato) [reply static-website-aws-csharp/Program.cs:17]: Actually on second thought, I'll do this as a follow-up, as they'll need to be added to Synced Folder as well, and I don't want to block on that. Easy enough to add post-release.

File: static-website-azure-csharp/Pulumi.yaml
Comment (nimbinatus) [inline static-website-azure-csharp/Pulumi.yaml:6]:
```suggestion
  description: A static website on Azure
```

File: static-website-azure-go/Pulumi.yaml
Comment (nimbinatus) [inline static-website-azure-go/Pulumi.yaml:6]:
```suggestion
  description: A static website on Azure
```

File: static-website-azure-python/Pulumi.yaml
Comment (nimbinatus) [inline static-website-azure-python/Pulumi.yaml:6]:
```suggestion
  description: A static website on Azure
```

File: static-website-azure-typescript/Pulumi.yaml
Comment (nimbinatus) [inline static-website-azure-typescript/Pulumi.yaml:6]:
```suggestion
  description: A static website on Azure
```

File: static-website-azure-yaml/Pulumi.yaml
Comment (nimbinatus) [inline static-website-azure-yaml/Pulumi.yaml:6]:
```suggestion
  description: A static website on Azure
```

File: static-website-aws-python/__main__.py
Comment (nimbinatus) [inline static-website-aws-python/__main__.py:17]: Generally, PEP8 would have one of the following adjustments. It is *not* a blocker, and I'm not going to comment on every single one here because that's up to you:
All arguments at the same indent level, individual lines:
```suggestion
bucket = aws.s3.Bucket(
    "bucket",
    acl="public-read",
    website=aws.s3.BucketWebsiteArgs(
        index_document=index_document,
        error_document=error_document,
    ))
```
OR
All arguments aligned with the opening delimiter:
```suggestion
bucket = aws.s3.Bucket("bucket",
                       acl="public-read",
                       website=aws.s3.BucketWebsiteArgs(
                           index_document=index_document,
                           error_document=error_document,
                       ))
```

File: static-website-aws-python/pyvenv.cfg
Comment (nimbinatus) [inline static-website-aws-python/pyvenv.cfg:1]: I wouldn't commit the `pyvenv.cfg` file. It generally isn't something we send in a repo (it lives in the `venv` directory if you're using the built-in virtual environment tool, which doesn't get committed), and someone will wonder why it appears. Pulumi creates the virtual environment based on the local Python environment, instead, and this file doesn't force it to use a specific version or anything.

File: static-website-azure-python/pyvenv.cfg
Comment (nimbinatus) [inline static-website-azure-python/pyvenv.cfg:1]: See prior note re:removal

File: static-website-gcp-python/pyvenv.cfg
Comment (nimbinatus) [inline static-website-gcp-python/pyvenv.cfg:1]: See prior note re:removal

File: static-website-azure-python/__main__.py
Comment (nimbinatus) [inline static-website-azure-python/__main__.py:4]: Are you specifically trying to use only built-in libraries? I generally only find `urllib` versus `requests` or `urllib3` when people are trying to restrict their dependency/security surface area.

Comment (nimbinatus) [reply static-website-azure-python/__main__.py:4]: Also, if you're only using one method or class, it's generally better practice to import just that need to avoid polluting your namespace, similar to the following:
```
from urllib.parse import urlparse
```

Finally, if you're going to import built-ins, generally PEP8 style has built-ins first (standard library modules), then third-party, then local (https://peps.python.org/pep-0008/#imports).

**Review by nimbinatus — CHANGES_REQUESTED** [review]
There's a couple places in the YAML files that need to get updated. The other stuff is more nits or questions.

File: static-website-aws-python/pyvenv.cfg
Comment (cnunciato) [reply static-website-aws-python/pyvenv.cfg:1]: Ah, good spot! I actually had a line in my code to delete these files, but it had a typo, and I didn't notice them sneaking through. Will remove. Thanks!

File: static-website-azure-python/__main__.py
Comment (cnunciato) [reply static-website-azure-python/__main__.py:4]: >Are you specifically trying to use only built-in libraries?

Ideally (and I guess in general) yes; assuming this works and isn't somehow problematic, I figure folks can opt into a different library if they want to.

Comment (cnunciato) [reply static-website-azure-python/__main__.py:4]: > PEP8 style has built-ins first (standard library modules), then third-party, then local

Ok thanks. So like this?

```python
from urllib.parse import urlparse
import pulumi
import pulumi_azure_native as azure_native
import pulumi_synced_folder as synced_folder
```

File: static-website-aws-python/__main__.py
Comment (cnunciato) [reply static-website-aws-python/__main__.py:17]: Ok great, thanks. I think I'll go with your first suggestion, then, as that's also what I get out of the box when I run `black` against these files -- and I personally find that version a bit more readable as well. 😄

**Comment by scottslowe** [comment]
@cnunciato In the `template` section, is there a `description` field supported for each key? I'm just wondering if it would be useful to have a prompt there to explain what each of the prompts is asking about.

File: static-website-azure-yaml/Pulumi.yaml
Comment (cnunciato) [inline static-website-azure-yaml/Pulumi.yaml]: Committed by mistake — will fix.

File: static-website-azure-python/__main__.py
Comment (nimbinatus) [reply static-website-azure-python/__main__.py]: > > Are you specifically trying to use only built-in libraries?
> 
> Ideally (and I guess in general) yes; assuming this works and isn't somehow problematic, I figure folks can opt into a different library if they want to.

Makes sense; it *is* a common practice if you're trying to reduce surface area. I probably would leave a comment in the line, then, just so newer Python folks aren't confused (most tutorials/self-guided learning systems use requests or urllib3; they very rarely use the built-in. If you want to explore why, check out https://realpython.com/urllib-request/#the-request-package-ecosystem, which has a very good writeup). 

If you add a comment, two spaces after the end of the import will make every linter happy :)

```python
import package  # Here's a comment
```

**Review by scottslowe — APPROVED** [review]

File: static-website-azure-python/__main__.py
Comment (nimbinatus) [reply static-website-azure-python/__main__.py]: > > PEP8 style has built-ins first (standard library modules), then third-party, then local
> 
> Ok thanks. So like this?
> 
> ```python
> from urllib.parse import urlparse
> import pulumi
> import pulumi_azure_native as azure_native
> import pulumi_synced_folder as synced_folder
> ```

Yep, that technically meets PEP8. A lot of folks follow a slightly modified version that puts the `from .. import ..` calls at the bottom in the same order. Let me add a couple packages to show you what I mean:

```python
import os
import sys

import pulumi
import pulumi_azure_native as azure_native
import pulumi_synced_folder as synced_folder

from urllib.parse import urlparse

from requests.exceptions import HTTPError
```

**However**, that gets into arguments as to which style to follow that are more pedantic than helpful. As long as you follow PEP8 and the code is readable, you're good.

File: static-website-aws-python/__main__.py
Comment (nimbinatus) [reply static-website-aws-python/__main__.py]: "Readability counts." 😉

**Comment by cnunciato** [comment]
@pulumi/platform So it looks like the tests are all passing for the static-website templates (🎉) except for GCP, which is complaining about the number of GlobalAddress resources being hit:

https://github.com/pulumi/templates/runs/8116507581?check_suite_focus=true#step:19:3182

The template uses the [GlobalAddress](https://www.pulumi.com/registry/packages/gcp/api-docs/compute/globaladdress/) resource to enable CDN functionality for users out of the box. Is there some way we can suppress these particular tests somehow?

File: static-website-gcp-yaml/Pulumi.yaml
Comment (AaronFriel) [inline static-website-gcp-yaml/Pulumi.yaml:39]: Nit: The next CLI release does not use the `.append` file, so any comments we want them to see should appear in both until we land #325.

**Review by AaronFriel — COMMENTED** [review]
@cnunciato left a comment about .append files.

File: static-website-gcp-yaml/Pulumi.yaml
Comment (cnunciato) [reply static-website-gcp-yaml/Pulumi.yaml:39]: Ah great, ok thanks @AaronFriel. Will drop these in there as well, thn.

**Pull request merged**
